makefile

```diff
@@ -423,10 +430,13 @@ else ifeq ($(call TRY_COMPILE, $(FIO_TLS_TEST_BEARSSL_EXT), "-lbearssl"), 0)
   $(info * Detected the BearSSL library, setting HAVE_BEARSSL)
 	FLAGS:=$(FLAGS) HAVE_BEARSSL
 	LINKER_LIBS_EXT:=$(LINKER_LIBS_EXT) bearssl
-else ifeq ($(call TRY_COMPILE, $(FIO_TLS_TEST_OPENSSL), "-lcrypto" "-lssl"), 0)
+else ifeq ($(call TRY_COMPILE, $(FIO_TLS_TEST_OPENSSL), $(shell pkg-config --cflags --libs openssl)), 0)
```

Comment:
Are you suggesting something like this?
```
   ...
else ifeq ([pkg-config check],0) && ([try-compile with pkg-config flags check])
   ...
else ifeq ([try-compile with fixed strings check])
   ...
```
In that case, for a user it would not be possible anymore to override these flags for `openssl` on the command line when the `[pkg-config check]` fails. I don't really need to, but it feels a bit inconsistent. (On a side note, I don't know how to do the `&&` in `GNU make`, sorry, I would break the structure of the `if`-`else`-chain.)

Reply:
I am sorry, I was referring the flags `OPENSSL_CFLAGS`, `OPENSSL_LDFLAGS` in the new version.

Reply:
Oh, no, this comment was here from before you fixed it. I loved your fix, more readable than my original suggestion.